internal/core/data/application/event.go

```diff
@@ -173,48 +173,60 @@ func DeleteEventsByDeviceName(deviceName string, dic *di.Container) errors.EdgeX
 }
 
 // AllEvents query events by offset and limit
-func AllEvents(offset int, limit int, dic *di.Container) (events []dtos.Event, err errors.EdgeX) {
+func AllEvents(offset int, limit int, dic *di.Container) (events []dtos.Event, totalCount uint32, err errors.EdgeX) {
 	dbClient := container.DBClientFrom(dic.Get)
 	eventModels, err := dbClient.AllEvents(offset, limit)
-	if err != nil {
-		return events, errors.NewCommonEdgeXWrapper(err)
+	if err == nil {
+		totalCount, err = dbClient.EventTotalCount()
 	}
-	events = make([]dtos.Event, len(eventModels))
-	for i, e := range eventModels {
-		events[i] = dtos.FromEventModelToDTO(e)
+	if err != nil {
+		return events, totalCount, errors.NewCommonEdgeXWrapper(err)
+	} else {
+		events = make([]dtos.Event, len(eventModels))
+		for i, e := range eventModels {
+			events[i] = dtos.FromEventModelToDTO(e)
+		}
+		return events, totalCount, nil
 	}
-	return events, nil
 }
 
 // EventsByDeviceName query events with offset, limit and name
-func EventsByDeviceName(offset int, limit int, name string, dic *di.Container) (events []dtos.Event, err errors.EdgeX) {
+func EventsByDeviceName(offset int, limit int, name string, dic *di.Container) (events []dtos.Event, totalCount uint32, err errors.EdgeX) {
 	if name == "" {
-		return events, errors.NewCommonEdgeX(errors.KindContractInvalid, "name is empty", nil)
+		return events, totalCount, errors.NewCommonEdgeX(errors.KindContractInvalid, "name is empty", nil)
 	}
 	dbClient := container.DBClientFrom(dic.Get)
 	eventModels, err := dbClient.EventsByDeviceName(offset, limit, name)
-	if err != nil {
-		return events, errors.NewCommonEdgeXWrapper(err)
+	if err == nil {
+		totalCount, err = dbClient.EventCountByDeviceName(name)
 	}
-	events = make([]dtos.Event, len(eventModels))
-	for i, e := range eventModels {
-		events[i] = dtos.FromEventModelToDTO(e)
+	if err != nil {
+		return events, totalCount, errors.NewCommonEdgeXWrapper(err)
+	} else {
```

Comment:
Remove `else` since not need due to above `return`.
This is a repeated pattern. Please fix every it is used.